src/kava.ts

```diff
@@ -875,16 +875,20 @@ class Kava extends BasePlugin {
   }
 
   private _getPosition(): number {
+    let currentTime = this.player.currentTime;
     if (this.player.isLive()) {
       if (!Number.isNaN(this.player.duration)) {
-        if (this.player.duration! - this.player.currentTime! < 1) {
+        if (this.player.duration! - currentTime! < 1) {
           return 0;
         }
-        return -(this.player.duration! - this.player.currentTime!);
+        return -(this.player.duration! - currentTime!);
       }
       return 0;
     }
-    return this._isFirstPlaying ? this.player.currentTime! || this.player.sources.startTime || 0 : this.player.currentTime!;
+    if (this.player.sources.seekFrom && currentTime) {
```

Comment:
well, i dont like the structure of the function, but i dont want to refactor the whole thing now.

i think this would be more readable

    let currentTime = this.player.currentTime;
    if (typeof this.player.sources.seekFrom === "number" && currentTime) {
      currentTime += this.player.sources.seekFrom;
    }

and in the use of currentTime in the lines above, keep the existing syntax of this.player.currentTime.

Reply:
done